ql/time/daycounters/actualactual.cpp

```diff
@@ -50,26 +50,53 @@ namespace QuantLib {
                                                    const Schedule& schedule) {
             // Process the schedule into an array of dates.
             Date issueDate = schedule.date(0);
-            Date firstCoupon = schedule.date(1);
-            Date notionalCoupon =
-                schedule.calendar().advance(firstCoupon,
-                                            -schedule.tenor(),
-                                            schedule.businessDayConvention(),
-                                            schedule.endOfMonth());
-
             std::vector<Date> newDates = schedule.dates();
-            newDates[0] = notionalCoupon;
-
-            //long first coupon
-            if (notionalCoupon > issueDate) {
-                Date priorNotionalCoupon =
-                    schedule.calendar().advance(notionalCoupon,
-                                                -schedule.tenor(),
-                                                schedule.businessDayConvention(),
-                                                schedule.endOfMonth());
-                newDates.insert(newDates.begin(),
-                                priorNotionalCoupon); //insert as the first element?
+
+            if (!schedule.isRegular(1))
```

Comment:
I would say 
```
if (!schedule.hasIsRegular() || !schedule.isRegular(1))
```
so that the case in which we don't have info works instead of throwing an exception.  Same below.

Reply:
Ah, thanks. Wasn't aware of the `schedule.has*` methods.